tools/src/test/java/com/nedap/archie/rules/evaluation/ParsedRulesEvaluationTest.java

```diff
@@ -622,5 +653,37 @@ public void stringLiterals() throws Exception {
 
     }
 
+    @Test
+    public void flattenedRules() throws IOException {
+        Archetype valueSet = parse("matches_valueset.adls");
+        Archetype parent = parse("termcodeparent.adls");
+        InMemoryFullArchetypeRepository repository = new InMemoryFullArchetypeRepository();
+        repository.addArchetype(parent);
+        repository.addArchetype(valueSet);
+        Flattener flattener = new Flattener(repository, BuiltinReferenceModels.getMetaModels(), FlattenerConfiguration.forOperationalTemplate());
+        OperationalTemplate opt = (OperationalTemplate) flattener.flatten(parent);
+        ExampleJsonInstanceGenerator generator = new ExampleJsonInstanceGenerator(BuiltinReferenceModels.getMetaModels(), "en");
+        Map<String, Object> exampleInstance = generator.generate(opt);
+        Cluster cluster = JacksonUtil.getObjectMapper().readValue(JacksonUtil.getObjectMapper().writeValueAsString(exampleInstance), Cluster.class);
+        //correct case first
+        RuleEvaluation ruleEvaluation = new RuleEvaluation(ArchieRMInfoLookup.getInstance(), JAXBUtil.getArchieJAXBContext(), opt);
+        DvCodedText codedText = (DvCodedText) cluster.itemAtPath("/items[1]/items[1]/value[1]");
+        codedText.setDefiningCode(new CodePhrase(new TerminologyId("local"), "at4"));
+        codedText.setValue("value 1");
+        EvaluationResult result = ruleEvaluation.evaluate(cluster, opt.getRules().getRules());
+        AssertionResult assertionResult = result.getAssertionResults().get(0);
+        assertEquals("ac3", assertionResult.getPathsConstrainedToValueSets().get("/items[id2]/items[id2]/value/defining_code"));
```

Comment:
```suggestion
        AssertionResult assertionResult = result.getAssertionResults().get(0);
        assertTrue(assertionResult.getResult());
        assertEquals("ac3", assertionResult.getPathsConstrainedToValueSets().get("/items[id2]/items[id2]/value/defining_code"));
```

You forgot the assert here. It also fails :-(